### ISSUE TITLE
Tables aligned on left on even pages when using two sided layout

### ISSUE THREAD

**Issue by Scarli177** [fmarotta/kaobook#66]
Hi, 

I'm using currently the two sided layout (twoside=true) for my master thesis and I have the problem that tables when they are on even pages, they align with the outer frame of the margin and not with the text. How can I fix this?

Thanks for this amazing template

**Comment by yordiak** [comment]
I think that he mentions in `kaobook` documentation that the `twoside` option is buggy. Does this problem occurs if `twoside=false`?

**Comment by Scarli177** [comment]
I just tried it, it doesn't. They align perfectly with the frame of the main text.

**Comment by fmarotta** [comment]
Hi, thanks for reaching out. Are you using the `\centering` command inside of the table environment?

**Comment by Scarli177** [comment]
No, i don't use the \centering. Normal tables (not the wide ones) are already centered on odd pages (margin on the right) but are aligned on the left when they appear on even pages. I attached some photos. 

![even_normal_table](https://user-images.githubusercontent.com/70272060/91440896-98c0af80-e86f-11ea-90fc-c4a615ef8b5b.PNG)
![even_raggedleft](https://user-images.githubusercontent.com/70272060/91440898-99f1dc80-e86f-11ea-9125-31c22ab8b7c5.PNG)
![even_raggedright](https://user-images.githubusercontent.com/70272060/91440899-99f1dc80-e86f-11ea-85ff-0d6900ea6cbc.PNG)
![odd_normal_table](https://user-images.githubusercontent.com/70272060/91440900-9a8a7300-e86f-11ea-99ad-be08ea6946fd.PNG)
![odd_raggedleft](https://user-images.githubusercontent.com/70272060/91440904-9b230980-e86f-11ea-9c15-0f0b8a0dcaa4.PNG)
![odd_raggedright](https://user-images.githubusercontent.com/70272060/91440906-9b230980-e86f-11ea-8fd7-4363ae292989.PNG)

**Comment by fmarotta** [comment]
Yep, it is probably a bug. For the next ~10 days I'll take a break from LaTeX (sorry about that), but I'll see what I can do soon enough after that.

**Comment by Scarli177** [comment]
Hi :-)

no hurry. Thanks for the help. I tried to figure out what the problem can be and saw that another table works just fine. The difference between the working table and the table I had the trouble with, is the caption. As  I don't need the caption for this table I just removed it. But it seems that the caption is pushing the tables at the right position. I'm not sure how to solve this globally, but I made it like this now for the tables where I don't need a caption:

\begin{tabular}{l|l}
$Q$ & Volume flow rate, \si{\meter ^{3} / \second} at \SI{1}{Pa}  \\
$k$ & flow coefficient \\
$\Delta P$ & Pressure difference across crack, \si{Pa} \\
$n$ & flow exponent 
\end{tabular}

**Comment by fmarotta** [comment]
Yes, the problem is indeed that if the caption is absent, the table is moved to the left by an amount equal to the space that the caption would have occupied, i.e. `\marginparwidth` + `\marginparsep`, and this happens only on even pages. I also discovered that figures are afflicted, too.

I have now added a way to check internally whether the table (or figure) has a caption, and, if it doesn't, I move it back to the correct position. This fix is implemented in the latest commit; I'll probably create a new release once it is confirmed that it works.
